.clang-tidy

```diff
@@ -1,27 +1,64 @@
 Checks: >
   -*,
-  readability-identifier-naming
+  performance-*,
+  readability-*,
```

Comment:
I would exclude
`-readability-braces-around-statements`
, because it will suggest to put braces into one-liners

```c++
 if (foo) return bar;
```

The `google-readability-braces-around-statements` is more useful. It allows the above, but will complain when things take more than a line
```c++
if (foo)
   return bar;
```

It emphasizes reducing vertical whitespace if possible.

Another example is `-readability-implicit-bool-conversion`. It complains when pointers are used in boolean expressions, but that is a very common and very readable technique. So

```c++
 if (thing) thing->foo();
```

... would need to become

```c++
if (thing != nullptr) thing->foo();
```

to satisfy the rule. Which is hiding intent and reduces readability.

.. there are a few others.

Reply:
I agree about -readability-implicit-bool-conversion.  I prefer readability-braces-around-statements.  It avoids the classic
```
if (cond) 
   a;
   b;
```
mistake.  I know others will disagree.

Reply:
both, `readability-braces-around-statements` and `google-readability-braces-around-statements` will complain about the example you mention and suggest to 

```c++
  if (cond) {
     a;
  }
  b;
```

The difference is, that `google-readability-braces-around-statements` will allow, if and only if, the result is in the same line, to not put braces around it. Because if there are braces around it, it will then have the formatter kick in and format it in three lines.

But for readability, typical one-liners (typically used in quick-out  `return`, `break` or `continue`) as in my example will not require braces.